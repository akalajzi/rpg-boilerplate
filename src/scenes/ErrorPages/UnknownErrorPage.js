// @flow strict

import React from 'react';
import styled from 'styled-components';

type Props = {
  title?: React$Node,
  description?: React$Node,
};

const ErrorPageWrapper = styled.div``;

const Title = styled.div``;

const Description = styled.div``;

const UnknownErrorPage = ({ title, description }: Props) => (
  <ErrorPageWrapper>
    <Title>{title}</Title>
    <Description>{description}</Description>
  </ErrorPageWrapper>
);

UnknownErrorPage.defaultProps = {
  title: 'Something went wrong',
  description: '',
};

export default UnknownErrorPage;
