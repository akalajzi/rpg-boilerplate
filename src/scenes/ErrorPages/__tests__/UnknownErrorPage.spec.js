// @flow strict

import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import UnknownErrorPage from '../UnknownErrorPage';

describe('#UnknownErrorPage', () => {
  test('render', () => {
    const component = renderer.create(
      <Router>
        <UnknownErrorPage />
      </Router>,
    );

    expect(component).toMatchSnapshot();
  });

  test('should render passed title and description', () => {
    const component = mount(
      <Router>
        <UnknownErrorPage title="Title" description="Description" />
      </Router>,
    );

    expect(component.find('UnknownErrorPage__Title').contains('Title')).toBe(true);
    expect(component.find('UnknownErrorPage__Description').contains('Description')).toBe(true);
  });
});
