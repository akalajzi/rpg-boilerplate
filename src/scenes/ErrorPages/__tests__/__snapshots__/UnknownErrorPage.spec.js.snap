// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#UnknownErrorPage render 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    Something went wrong
  </div>
  <div
    className=""
  >
    
  </div>
</div>
`;
